colcon-build/action.yaml

```diff
@@ -66,10 +66,16 @@ runs:
           --mixin coverage-gcc coverage-pytest compile-commands
       shell: bash
 
+    - name: Determine CPU architecture
+      id: determine-architecture
+      run: |
+        echo ::set-output name=arch::$(uname -m)
+      shell: bash
+
     - name: Cache build artifacts
       uses: actions/cache@v2
       with:
         path: |
           ./build
           ./install
-        key: build-${{ github.sha }}
+        key: build-${{ runner.os }}-${{ steps.determine-architecture.outputs.arch }}-${{ github.sha }}
```

Comment:
Is it impossible to use `runner.arch`?
https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context

Reply:
Sounds good! I will use it.

Reply:
Changed: 9ccbb23